lib/game.ex

```diff
@@ -15,4 +16,29 @@ defmodule TicTacToe.Game do
     Map.replace!(game, :players, Game.get_players(input))
   end
 
+  def add_board(game) do
+    Map.replace!(game, :board, Board.new(3))
+  end
+
+  def make_move(game, move, input \\ CLI) do
+    [ current_player | _tail ] = game.players
+    cond do
+      !Board.space_is_available(game.board.spaces, move) ->
+        input.print "That space isn't available, please choose an open space between 1 and #{length(game.board.spaces)}"
+        new_move = input.get_player_move(current_player.name)
+        make_move(game, new_move, input)
+      Board.space_is_available(game.board.spaces, move) ->
+        new_board = Board.update_at(game.board.spaces, move, current_player.marker)
+        update_in(game, [Access.key(:board), Access.key(:spaces)], &(&1 = new_board))
+    end
+  end
```

Comment:
this conditional switch is looking like a lot, maybe we can break this down further / extract functions out

Reply:
```Elixir
  def update(board, game) do
    update_in(game, [Access.key(:board), Access.key(:spaces)], &(&1 = board))
  end

  def next_move(game) do
    Game.current_player(game)
      |> CLI.get_player_move
      |> Board.validate_move(game.board)
      |> Game.make_move(game)
  end

  def make_move(-1, game) do
    Game.next_move(game)
  end

  def make_move(move, game) do
    { current_player, board } = { Game.current_player(game), game.board }
    Board.update_at(board.spaces, move, current_player.marker)
      |> Game.update(game)
  end
```

I extracted the validation/conditional logic to trim this down ✂️